Add full domain support

gerrod3 opened the pull request:
fixes: #1534
https://issues.redhat.com/browse/PULP-344

dralley commented on docs/admin/learn/domain.md:
```diff
@@ -0,0 +1,54 @@
+
+!!! note
+
+    The Flatpak endpoints will only work within the default domain, even with domains enabled.
```
Is that a temporary or permanent limitation?

gerrod3 replied:
Temporary, I hope. I need to do more research into flatpak and how it works (the docs aren't very helpful about using flatpak with an oci-registry) to see if we can somehow add the domain-name to the urls that flatpak uses (currently `/index/static/` and `/index/dynamic/`

dralley commented:
@gerrod3 Can you give me some idea of what changed since the last time I looked it over (when that comment was made)?

dralley commented:
Or should I just do a full look through again (if a lot has changed)

gerrod3 commented:
@dralley The only thing I changed was the tests. I fixed them for the CI (properly take into the account the different settings the CI is tested under).
